crates/valence_command/src/handler.rs

```diff
@@ -0,0 +1,329 @@
+use std::collections::HashMap;
+
+use bevy_app::{App, Plugin, PostStartup, Update};
+use bevy_ecs::change_detection::{Res, ResMut};
+use bevy_ecs::event::{Event, EventReader, EventWriter};
+use bevy_ecs::prelude::{Entity, Resource};
+use bevy_ecs::system::Query;
+use petgraph::algo::all_simple_paths;
+use petgraph::dot::Dot;
+use petgraph::prelude::NodeIndex;
+use petgraph::Graph;
+use valence_server::event_loop::PacketEvent;
+use valence_server::protocol::packets::play::CommandExecutionC2s;
+
+use crate::arg_parser::ArgLen;
+use crate::command_graph::{
+    parser_len, parser_valid_for, CommandEdgeType, CommandGraphBuilder, CommandNode, NodeData,
+};
+use crate::{Command, CommandRegistry, CommandScopeRegistry, CommandTypingEvent};
+use crate::command_scopes::CommandScopes;
+
+pub struct CommandHandler<T>
+where
+    T: Command + Resource + Clone,
+{
+    command: T,
+}
+
+impl<T> CommandHandler<T>
+where
+    T: Command + Resource + Clone,
+{
+    pub fn from_command(command: T) -> Self {
+        CommandHandler { command }
+    }
+
+    pub fn command_name() -> String {
+        T::name()
+    }
+}
+
+#[derive(Resource)]
+struct CommandResource<T: Command> {
+    command: T,
+    executables: HashMap<NodeIndex, fn(Vec<String>) -> T::CommandExecutables>,
+}
+
+impl<T: Command> CommandResource<T> {
+    pub fn new(command: T) -> Self {
+        CommandResource {
+            command,
+            executables: HashMap::new(),
+        }
+    }
+}
+
+#[derive(Event)]
+pub struct CommandExecutionEvent<T>
+where
+    T: Command,
+    T::CommandExecutables: Send + Sync + 'static,
+{
+    pub result: T::CommandExecutables,
+    pub executor: Entity,
+}
+
+impl<T> Plugin for CommandHandler<T>
+where
+    T: Command + Resource + Clone,
+{
+    fn build(&self, app: &mut App) {
+        println!("Registering command: {}", Self::command_name());
+
+        app.add_event::<CommandTypingEvent<T>>()
+            .add_event::<CommandExecutionEvent<T>>()
+            .insert_resource(CommandResource::new(self.command.clone()))
+            .add_systems(Update, command_event_system::<T>)
+            .add_systems(PostStartup, command_startup_system::<T>);
+
+        println!("Registered command: {}", Self::command_name());
```

Comment:
Bevy already logs when plugins are loaded. (if this is just a temp thing that's fine)